src/main/java/org/example/repository/CentralRepository.java

```diff
@@ -0,0 +1,79 @@
+package org.example.repository;
+
+import org.example.entity.*;
+import org.example.exception.NoTenantException;
+import org.example.exception.OwnerHasNoPropertyException;
+import org.example.exception.RentingPropertyWithoutOwnerException;
+import org.example.exception.TenantIsAlreadySetInOtherPropertyException;
+import org.example.interfaces.IRepo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class CentralRepository implements IRepo<RentData> {
+
+    private final List<RentData> data;
+    private static CentralRepository INSTANCE;
```

Comment:
You are trying to implement singleton here, and still we use CAPITAL names only for constants, instance in singletons are still lowercased.